:bomb: `Bug`: Runtime Error: Nil Pointer Dereference with stateFile Validation Failed.

### What happened?

When I tested the gitops workflow after execute:

`./dtm init -f config-gitops.yaml `

Then I got the error below:

```sh
2022-05-20 13:36:06 ✖ [ERROR]  Config file validation failed: stateFile is empty.
2022-05-20 13:36:06 ℹ [INFO]  Using dir <.devstream> to store plugins.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1055a3f98]

goroutine 1 [running]:
github.com/devstream-io/devstream/internal/pkg/pluginmanager.DownloadPlugins(0x0)
        github.com/devstream-io/devstream/internal/pkg/pluginmanager/manager.go:30 +0x158
main.initCMDFunc(0x105c2b620, {0x1400008bce0, 0x0, 0x2})
        github.com/devstream-io/devstream/cmd/devstream/init.go:25 +0x15c
github.com/spf13/cobra.(*Command).execute(0x105c2b620, {0x1400008bcc0, 0x2, 0x2})
        github.com/spf13/cobra@v1.4.0/command.go:860 +0xa04
github.com/spf13/cobra.(*Command).ExecuteC(0x105c2c020)
        github.com/spf13/cobra@v1.4.0/command.go:974 +0x7b8
github.com/spf13/cobra.(*Command).Execute(0x105c2c020)
        github.com/spf13/cobra@v1.4.0/command.go:902 +0x30
main.main()
        github.com/devstream-io/devstream/cmd/devstream/main.go:105 +0x30
```

### How to reproduce?

**step 1:**

```sh
curl -o config-gitops.yaml https://raw.githubusercontent.com/devstream-io/devstream/main/examples/gitops.yaml
curl -o variables-gitops.yaml https://raw.githubusercontent.com/devstream-io/devstream/main/examples/variables-gitops.yaml
curl -o tools-gitops.yaml https://raw.githubusercontent.com/devstream-io/devstream/main/examples/tools-gitops.yaml
```

**step 2:**

```sh
make build -j8
./dtm init -f config-gitops.yaml
```

### DevStream version (to find out the version run: dtm version)

latest

--- a/internal/pkg/configloader/config.go
+++ b/internal/pkg/configloader/config.go
@@ -29,7 +29,7 @@ type Config struct {
 type ConfigFile struct {
 	VarFile  string `yaml:"varFile"`
 	ToolFile string `yaml:"toolFile"`
-	State    *State
+	State    *State `yaml:"state"`
 }
 
 type State struct {
@@ -39,11 +39,11 @@ type State struct {
 
 type StateConfigOptions struct {
 	// for s3 backend
-	Bucket string
-	Region string
-	Key    string
+	Bucket string `yaml:"bucket"`
+	Region string `yaml:"region"`
+	Key    string `yaml:"key"`
 	// for local backend
-	StateFile string
+	StateFile string `yaml:"stateFile"`
 }
 
 // Tool is the struct for one section of the DevStream configuration file.

--- a/internal/pkg/configloader/validation.go
+++ b/internal/pkg/configloader/validation.go
@@ -5,6 +5,9 @@ import (
 	"strings"
 
 	"k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/devstream-io/devstream/internal/pkg/backend/local"
+	"github.com/devstream-io/devstream/pkg/util/log"
 )
 
 func validateConfig(config *Config) []error {
@@ -90,9 +93,11 @@ func validateConfigFile(c *ConfigFile) []error {
 	if c.State == nil {
 		errors = append(errors, fmt.Errorf("state config is empty"))
 	} else {
+		log.Debugf("Got Backend from config: %s", c.State.Backend)
 		if c.State.Backend == "local" {
 			if c.State.Options.StateFile == "" {
-				errors = append(errors, fmt.Errorf("stateFile is empty"))
+				log.Debugf("The stateFile has not been set, default value %s will be used.", local.DefaultStateFile)
+				c.State.Options.StateFile = local.DefaultStateFile
 			}
 		} else if c.State.Backend == "s3" {
 			if c.State.Options.Bucket == "" {
@@ -104,6 +109,8 @@ func validateConfigFile(c *ConfigFile) []error {
 			if c.State.Options.Key == "" {
 				errors = append(errors, fmt.Errorf("state s3 Key is empty"))
 			}
+		} else {
+			errors = append(errors, fmt.Errorf("backend type error"))
 		}
 	}
 

